java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/WeightActivity.java

```diff
@@ -0,0 +1,162 @@
+package com.microsoft.hsg.android.hvsample;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.android.simplexml.HealthVaultApp;
+import com.microsoft.hsg.android.simplexml.ShellActivity;
+import com.microsoft.hsg.android.simplexml.client.HealthVaultClient;
+import com.microsoft.hsg.android.simplexml.client.RequestCallback;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.request.ThingRequestGroup2;
+import com.microsoft.hsg.android.simplexml.methods.getthings3.response.ThingResponseGroup2;
+import com.microsoft.hsg.android.simplexml.things.thing.Thing2;
+import com.microsoft.hsg.android.simplexml.things.types.types.PersonInfo;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+import com.microsoft.hsg.android.simplexml.things.types.weight.Weight;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+import android.widget.AdapterView.OnItemSelectedListener;
+
+public class WeightActivity extends Activity {
+
+	private HealthVaultApp service;
+    private HealthVaultClient hvClient;
+    private Record currentRecord;
+    
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.weight);
+        service = HealthVaultApp.getInstance();
+        hvClient = new HealthVaultClient();
+        
+        Button weightsBtn = (Button) findViewById(R.id.addWeight);
+        final EditText editText = (EditText) findViewById(R.id.weightInput);
```

Comment:
I haven't seen many other instances of "final" in this PR but if we are going to be good about using it we will need to be consistent and use it all places where it applies.